### PR TITLE
Enabling bus support for hello_events.c

### PR DESCRIPTION
**Pull request by trenduchintala**
Allows hello_events.c to run on a specific bus. 

Usage:

hello_events -B 03
hello_events --bus 0x03

### CODE REVIEW

File: samples/hello_events.c [hunk 1]
```diff
@@ -129,29 +253,32 @@ int main(int argc, char *argv[])
...
-	res = fpgaEnumerate(&filter, 1, &fpga_device_token, 1, &num_matches);
-	ON_ERR_GOTO(res, out_destroy_tok, "enumerating accelerators");
+	res = find_fpga(&fpga_device_token, &num_matches);
+	ON_ERR_GOTO(res, out_exit, "finding FPGA accelerator");
 
 	if (num_matches < 1) {
-		fprintf(stderr, "accelerator not found.\n");
-		res = fpgaDestroyProperties(&filter);
-		ON_ERR_GOTO(res, out_destroy_tok, "injecting error");
+		res = FPGA_NOT_FOUND;
```
Comment (asgibbons) [inline samples/hello_events.c]: There should probably be a goto here or a return.

File: samples/hello_events.c [hunk 2]
```diff
@@ -115,12 +131,120 @@ static fpga_result inject_ras_fatal_error(fpga_token token, uint8_t err)
...
 }
 
+/*
+ * Parse command line arguments
+ */
+#define GETOPT_STRING "B:"
+fpga_result parse_args(int argc, char *argv[])
+{
+	struct option longopts[] = {
+		{"bus", required_argument, NULL, 'B'}
```
Comment (tswhison) [inline samples/hello_events.c]: longopts needs to be NULL-terminated.  ie
{ "bus", required_argument, NULL, 'B' },
{ NULL, 0, NULL, 0 }

File: samples/hello_events.c [hunk 3]
```diff
@@ -115,12 +131,120 @@ static fpga_result inject_ras_fatal_error(fpga_token token, uint8_t err)
...
+fpga_result get_bus_info(fpga_token tok, struct bdf_info *finfo){
+	fpga_result res = FPGA_OK;
+	fpga_properties props;
+	res = fpgaGetProperties(tok, &props);
+	ON_ERR_GOTO(res, out, "reading properties from Token");
+
+	res = fpgaPropertiesGetBus(props, &finfo->bus);
+	ON_ERR_GOTO(res, out_destroy, "Reading bus from properties");
+	
+	if(res != FPGA_OK){
```
Comment (tswhison) [inline samples/hello_events.c]: unnecessary check.  This is handled by the ON_ERR_GOTO above.

**Comment by buie1** [comment]
@OPAE/api Can we get this merged in.

File: samples/hello_events.c [hunk 4]
```diff
@@ -115,12 +131,117 @@ static fpga_result inject_ras_fatal_error(fpga_token token, uint8_t err)
...
+	if (-1 != config.target.bus) {
+		res = fpgaPropertiesSetBus(filter, config.target.bus);
+		ON_ERR_GOTO(res, out_destroy, "setting bus");
+	}
+		
+	res= fpgaEnumerate(&filter, 1, fpga, 1, num_matches);
+	ON_ERR_GOTO(res, out_destroy, "enumerating FPGAs");
+
+out_destroy:
+	res = fpgaDestroyProperties(&filter);
```
Comment (tswhison) [inline samples/hello_events.c]: If one of the ON_ERR_GOTO statements on 185, 188, 193, or 197 is taken, then the failure code stored in res is overwritten by line 200.  Could you create a 
fpga_result dres;

dres = fpgaDestroyProperties
return (res == FPGA_OK) ? dres : res;

**Comment by buie1** [comment]
awaiting approval from @opae/api. Is there anything else we need for this?

**Pull request merged**